include/vara/Feature/Feature.h

```diff
@@ -105,14 +108,23 @@ class Feature {
     Relationships.push_back(std::move(Relationship));
   }
 
+  [[nodiscard]] std::optional<std::unique_ptr<Location>> getLocation() const {
+    return Loc ? std::optional{std::make_unique<Location>(
+                     Loc->get()->getPath(), Loc->get()->getStart(),
+                     Loc->get()->getEnd())}
+               : std::nullopt;
+  }
+
   [[nodiscard]] virtual std::string toString() const;
 };
 
 /// Options without arguments.
 class BinaryFeature : public Feature {
 
 public:
-  BinaryFeature(string Name, bool Opt) : Feature(std::move(Name), Opt) {}
+  BinaryFeature(string Name, bool Opt,
+                std::optional<std::unique_ptr<Location>> Loc)
```

Comment:
I think we could default construct the optional or create a second ctor that takes no location. This would make it easier for callers because they do not have to pass in std::nullopts if the location is missing.